src/app/clusters/ota-requestor/DefaultOTARequestorStorage.cpp

```diff
@@ -134,6 +134,40 @@ CHIP_ERROR DefaultOTARequestorStorage::LoadUpdateToken(MutableByteSpan & updateT
     return Load(DefaultStorageKeyAllocator::OTAUpdateToken(), updateToken);
 }
 
+CHIP_ERROR DefaultOTARequestorStorage::StoreCurrentUpdateState(OTAUpdateStateEnum currentUpdateState)
+{
+    return mPersistentStorage->SyncSetKeyValue(DefaultStorageKeyAllocator::OTACurrentUpdateState(), &currentUpdateState,
+                                               sizeof(currentUpdateState));
+}
+
+CHIP_ERROR DefaultOTARequestorStorage::LoadCurrentUpdateState(OTAUpdateStateEnum & currentUpdateState)
+{
+    uint16_t size = static_cast<uint16_t>(sizeof(currentUpdateState));
+    return mPersistentStorage->SyncGetKeyValue(DefaultStorageKeyAllocator::OTACurrentUpdateState(), &currentUpdateState, size);
+}
+
+CHIP_ERROR DefaultOTARequestorStorage::ClearCurrentUpdateState()
+{
+    return mPersistentStorage->SyncDeleteKeyValue(DefaultStorageKeyAllocator::OTACurrentUpdateState());
+}
+
+CHIP_ERROR DefaultOTARequestorStorage::StoreTargetVersion(uint32_t targetVersion)
+{
+    return mPersistentStorage->SyncSetKeyValue(DefaultStorageKeyAllocator::OTATargetVersion(), &targetVersion,
+                                               sizeof(targetVersion));
+}
+
+CHIP_ERROR DefaultOTARequestorStorage::LoadTargetVersion(uint32_t & targetVersion)
```

Comment:
Is OTA requestor expected to validate if the new images version matches the version that was announced by OTA provider? I wonder if after applying the update it's not too late to validate that which would indicate that the target version doesn't need to be stored in KVS.

Reply:
I thought it might be a good idea for OTARequestor to validate version after first boot into the new image (especially helpful for Linux verification). Specifically for Linux, there wouldn't be an opportunity to validate after "applying" as it implies a reboot immediately after apply.